incapsula/resource_siem_connnection_s3_test.go

```diff
@@ -0,0 +1,81 @@
+package incapsula
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"log"
+	"os"
+	"testing"
+)
+
+//for tests without connectivity check put env variable: TESTING_PROFILE=true
+
+const s3SiemConnectionResourceType = "incapsula_siem_connection_s3"
+const s3SiemConnectionResourceName = "test_acc"
+const s3SiemConnectionResource = s3SiemConnectionResourceType + "." + s3SiemConnectionResourceName
+
+const siemConnectionS3StorageTypeValue = "CUSTOMER_S3"
+
+var s3SiemConnectionName = "SIEMCONNECTIONS3" + RandomLetterAndNumberString(10)
+
+func TestAccS3SiemConnection_Basic(t *testing.T) {
+	log.Printf("========================BEGIN TEST========================")
+	log.Printf("[DEBUG]Running test resource_s3_siem_connection.go.TestAccS3SiemConnection_Basic")
+
+	if v := os.Getenv("TESTING_PROFILE"); v != "" {
+		log.Printf("[DEBUG]TESTING_PROFILE environment variable is provided, test is skipped")
+		return
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccIncapsulaSiemConnectionDestroy(s3SiemConnectionResourceType),
+		Steps: []resource.TestStep{
+			{
+				Config: getAccIncapsulaS3SiemConnectionConfigBasic(),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckIncapsulaSiemConnectionExists(s3SiemConnectionResource),
+					resource.TestCheckResourceAttr(s3SiemConnectionResource, "connection_name", s3SiemConnectionName),
+					resource.TestCheckResourceAttr(s3SiemConnectionResource, "storage_type", siemConnectionS3StorageTypeValue),
+				),
+			},
+			{
+				ResourceName:      s3SiemConnectionResource,
+				ImportState:       true,
+				ImportStateVerify: false,
+				ImportStateIdFunc: testACCStateSiemConnectionID(s3SiemConnectionResourceType),
+			},
+		},
+	})
+}
+
```

Comment:
need to add a test for update operation like here:
https://github.com/imperva/terraform-provider-incapsula/blob/account_user/incapsula/resource_account_role_test.go#L40-L64